pytroll_collectors/fsspec_to_message.py

```diff
@@ -25,31 +26,50 @@ def create_message(fs, file, subject, metadata=None):
 def _create_message_metadata(fs, file):
     """Create a message to send."""
     loaded_fs = json.loads(fs)
-    uri = _create_uri(file, loaded_fs)
-    uid = _create_uid_from_uri(uri, loaded_fs)
+    uid = _create_uid(file, loaded_fs)
+    uri = _create_uid_from_uri(uid, loaded_fs)
     base_data = {'filesystem': loaded_fs, 'uri': uri, 'uid': uid}
     base_data.update(file.get('metadata', dict()))
     return base_data
 
 
-def _create_uri(file, loaded_fs):
+def _create_uid(file, loaded_fs):
     protocol = loaded_fs["protocol"]
     if protocol == 'abstract' and 'zip' in loaded_fs['cls']:
         protocol = 'zip'
-    uri = protocol + '://' + file['name']
-    return uri
+    netloc = create_netloc(loaded_fs)
+    uid = protocol + '://' + netloc + file['name']
+    return uid
 
 
-def _create_uid_from_uri(uri, loaded_fs):
-    uid = uri
+def _create_uid_from_uri(uid, loaded_fs):
+    uri = uid
     if 'target_protocol' in loaded_fs:
-        uid += '::' + loaded_fs['target_protocol'] + '://' + (loaded_fs.get('fo') or loaded_fs['args'][0])
-    return uid
+        netloc = create_netloc(loaded_fs['target_options'])
+        uri += '::' + loaded_fs['target_protocol'] + '://' + netloc + (loaded_fs.get('fo') or loaded_fs['args'][0])
+    return uri
+
+
+def create_netloc(ssh):
+    """Create the netloc from ssh parameters like `host`, `port`, `username`, `password`."""
+    host = ssh.get("host")
+    if host is None:
+        return ""
+    username = ssh.get("username")
+    password = ssh.get("password")
```

Comment:
I hope this doesn't imply that the credentials are passed in plain text in the messages everyone within the network can see...?